examples/mnist.py

```diff
@@ -123,16 +123,17 @@ def create_iter_functions(dataset, output_layer,
     batch_slice = slice(batch_index * batch_size,
                         (batch_index + 1) * batch_size)
 
-    objective = lasagne.objectives.Objective(output_layer,
-        loss_function=lasagne.objectives.categorical_crossentropy)
+    output = lasagne.layers.get_output(output_layer, X_batch)
+    loss_train = lasagne.objectives.categorical_crossentropy(output, y_batch)
+    loss_train = lasagne.objectives.aggregate(loss_train, 'mean')
```

Comment:
I don't really see the value here of using `aggregate` instead of just writing `loss_train = T.mean(loss_train)`. If I understand correctly it does exactly the same thing anyway. Sure, `aggregate` is more flexible, but since we just want the mean here, why not write that directly so people realise everything is just passing around Theano expressions that they can do with as they please.